lib/modules/platform/gitlab/index.ts

```diff
@@ -538,20 +538,28 @@ async function ignoreApprovals(pr: number): Promise<void> {
       }[]
     >(url);
 
+    const ruleName = 'renovateIgnoreApprovals';
+
     const existingAnyApproverRule = rules?.find(
       ({ rule_type }) => rule_type === 'any_approver'
     );
+    const existingRegularApproverRules = rules?.filter(
+      ({ rule_type, name }) => rule_type !== 'any_approver' && name !== ruleName
+    );
+
+    await Promise.all(
+      existingRegularApproverRules?.map(async (rule) => {
+        await gitlabApi.deleteJson(`${url}/${rule.id}`);
+      })
```

Comment:
use `all` from `util/promises`to limit concurrency.

https://github.com/renovatebot/renovate/blob/45c2d13f8a1dfffe432e3cc02892119cca7131ef/lib/util/promises.ts#L38-L52